File: client/clojure/src/cdisc_library_api/specs/cdashig_scenario_field.clj
(ns cdisc-library-api.specs.cdashig-scenario-field
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.cdashig-scenario-field-links :refer :all]
            )
  (:import (java.io File)))


(def cdashig-scenario-field-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :definition) string?
   (ds/opt :questionText) string?
   (ds/opt :prompt) string?
   (ds/opt :completionInstructions) string?
   (ds/opt :implementationNotes) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :mappingInstructions) string?
   (ds/opt :core) string?
   (ds/opt :_links) cdashig-scenario-field-links-spec
   })

(def cdashig-scenario-field-spec
  (ds/spec
    {:name ::cdashig-scenario-field
     :spec cdashig-scenario-field-data}))

File: client/scalaz/src/main/scala/org/openapitools/client/api/CdashigScenarioField.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CdashigScenarioField._

case class CdashigScenarioField (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
definition: Option[String],
questionText: Option[String],
prompt: Option[String],
completionInstructions: Option[String],
implementationNotes: Option[String],
simpleDatatype: Option[String],
mappingInstructions: Option[String],
core: Option[String],
links: Option[CdashigScenarioFieldLinks])

object CdashigScenarioField {
  import DateTimeCodecs._

  implicit val CdashigScenarioFieldCodecJson: CodecJson[CdashigScenarioField] = CodecJson.derive[CdashigScenarioField]
  implicit val CdashigScenarioFieldDecoder: EntityDecoder[CdashigScenarioField] = jsonOf[CdashigScenarioField]
  implicit val CdashigScenarioFieldEncoder: EntityEncoder[CdashigScenarioField] = jsonEncoderOf[CdashigScenarioField]
}
